### PR TITLE
[#153] regex facts

### PR DESCRIPTION
**Pull request by Samuel-Campbell**
- Regex demands and facts
- Binary model --> dictionary['filename']: vector
- every fact is a 0 or a 1.
0 indicates it was not mentioned
1 indicates it was mentioned

lease_was_paid and lease_was_not_paid are 2 seperate columns

### CODE REVIEW

**Review by dodoels — CHANGES_REQUESTED** [review]
Remove data from PR

File: src/ml_service/feature_extraction/regex/regex_lib.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        ("tenant_lease_fixed", re.compile(r".*((bail.*(reconduit.*)?(terminant.*)?((\d+?\w*?\s+?|)(janvier|février|mars|avril|mai|juin|juillet|août|septembre|octobre|novembre|décembre)(\s+\d{2,4}|))((.+au|.*terminant).*(\d+?\w*?\s+?|)(janvier|février|mars|avril|mai|juin|juillet|août|septembre|octobre|novembre|décembre)(\s+\d{2,4}|))?)|(fixation\sde\sloyer))"), re.IGNORECASE),
+        ("tenant_monthly_payment", re.compile(r"..*loyer\smensuel.*\b(\d{1,3}(\s\d{3}|,\d{2})*)+(\$|\s\$).*"), re.IGNORECASE),
+        ("tenant_owes_rent", re.compile(r"(.*preuve.+locateur(s)?.+non-paiement.+loyer)?.*(locataire(s)?.+(doi(vent|t))((.+somme\sde)|(total))?.+([\d\s,]+)(\$|\s\$)|locateur.+créance.+\b(\d{1,3}(\s\d{3}|,\d{2})*)+(\$|\s\$).+loyers\simpayés)|(.*paiement.*arriérés.+loyer.+\b(\d{1,3}(\s\d{3}|,\d{2})*))"), re.IGNORECASE),
+        ("tenant_rent_not_paid_more_3_weeks", re.compile(r".*locataire(s)?.+retard.+plus.+((trois semaines)|(trois \(3\) semaines)).+(paiement\sdu\sloyer)"), re.IGNORECASE),
+        ("tenant_violence", re.compile(r".*(raison.+)?.*(viol(ent|ence))"), re.IGNORECASE),
+        #("tenant_request_cancel_lease", re.compile(r"")),
```
Comment (TaimoorRana) [inline src/ml_service/feature_extraction/regex/regex_lib.py]: please remove all commented code

File: src/ml_service/feature_extraction/regex/data/demands/0-tenant_eviction.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
+Le locateur demande la résiliation du bail et l'expulsion des locataires
```
Comment (TaimoorRana) [inline src/ml_service/feature_extraction/regex/data/demands/0-tenant_eviction.txt:1]: Why are we uploading all of these?

File: src/ml_service/feature_extraction/regex/regex_lib.py (hunk 1 above)
Comment (Samuel-Campbell) [reply src/ml_service/feature_extraction/regex/regex_lib.py]: no

File: src/ml_service/feature_extraction/regex/data/demands/0-tenant_eviction.txt (hunk 2 above)
Comment (Samuel-Campbell) [reply src/ml_service/feature_extraction/regex/data/demands/0-tenant_eviction.txt:1]: yeah we will upload them on the server

Comment (Samuel-Campbell) [reply src/ml_service/feature_extraction/regex/data/demands/0-tenant_eviction.txt:1]: i will need to rebase

File: src/ml_service/feature_extraction/regex/regex_lib.py (hunk 1 above)
Comment (TaimoorRana) [reply src/ml_service/feature_extraction/regex/regex_lib.py]: yes.

File: src/ml_service/feature_extraction/regex/data/demands/0-tenant_eviction.txt (hunk 2 above)
Comment (TaimoorRana) [reply src/ml_service/feature_extraction/regex/data/demands/0-tenant_eviction.txt:1]: But why do we need to upload them?

**Review by TaimoorRana — CHANGES_REQUESTED** [review]
please remove all commented code. We need to keep the code base as clean as possible.

**Review by TaimoorRana — DISMISSED** [review]

File: src/ml_service/feature_extraction/regex/regex_lib.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class RegexLib:
+    regex_demands = [
+        ("tenant_eviction", re.compile(r".*(locat(eur(s)?|rice(s)?)).*(expulsion|éviction).*locataire(s)?"), re.IGNORECASE),
+        ("landlord_lease_termination", re.compile(r".*(locat(eur(s)?|rice(s)?)).*.+résiliation.+bail"), re.IGNORECASE),
+        ("tenant_cover_rent", re.compile(r".*(demande(nt)?.+)?(locat(eur(s)?|rice(s)?))(s)?(.+demande(nt)?)?.+recouvrement\s(du|de)\sloyer"), re.IGNORECASE),
+        ("paid_judicial_fees", re.compile(r".*(locat(eur(s)?|rice(s)?))\sdemande(nt)?.+(frais\sjudiciaires)"), re.IGNORECASE),
+        ("landlord_money_cover_rent", re.compile(r".*(locat(eur(s)?|rice(s)?)).+(demand(e|ent))?.+recouvrement.+\sloyer"), re.IGNORECASE),
+        ("landlord_fix_rent", re.compile(r".*(locat(eur(s)?|rice(s)?)).+(demand(e|ent)).+(fix(er|ation)).+loyer")),
```
Comment (TaimoorRana) [inline src/ml_service/feature_extraction/regex/regex_lib.py:12]: why are we not ignoring case for this one?

File: src/ml_service/feature_extraction/regex/regex_lib.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        ("tenant_rent_paid_before_hearing", re.compile(r".*locataire(s)?.*payé.*loyer.*(dû le jour|avant).+(audience).*"), re.IGNORECASE),
+        ("landlord_inspector_fees", re.compile(r".*(locat(eur(s)?|rice(s)?)).+(frais\sde\sdépistage).*"), re.IGNORECASE),
+        ("landlord_relocation_indemnity_fees", re.compile(r".*(locat(eur|rice)(s))?.+réclame.+indemnité.*\b(\d{1,3}(\s\d{3}|,\d{2})*)+(\$|\s\$).*"), re.IGNORECASE),
+        ("tenant_continuous_late_payment", re.compile(r".*(fréquen(ce|ts)).*(retard(s)?)?.*(article\s1971)?"), re.IGNORECASE),
+        ("tenant_landlord_agreement", re.compile(r".*entente.+(entre\sles\sdeux\sparties).*"), re.IGNORECASE),
+        ("landlord_pays_indemnity", re.compile(r".*dédommage.+locataire(s)?.*"), re.IGNORECASE),
+        ("apartment_impropre", re.compile(r".*(logement(était\s)?(.+impropre|infesté).+l'habitation|(bon\sétat.+propreté)).*"), re.IGNORECASE),
+        ("landlord_is_a_rapist", re.compile(r".*violentée|violent(e)?|menaçant(e)?.*"), re.IGNORECASE),
+        ("landlord_is_violent", re.compile(r".*violentée|violent(e)?|menaçant(e)?.*"), re.IGNORECASE),
```
Comment (TaimoorRana) [inline src/ml_service/feature_extraction/regex/regex_lib.py:65]: The 2 above are identical

File: src/ml_service/feature_extraction/regex/regex_lib.py (hunk 3 above)
Comment (Samuel-Campbell) [reply src/ml_service/feature_extraction/regex/regex_lib.py:12]: forgot to

**Comment by Taimoorrana1** [comment]
Closed as it was merged with #183

**Comment by TaimoorRana** [comment]
Closed as it was merged with #183

**Pull request closed without merging**